torch/_inductor/scheduler.py

```diff
@@ -2031,8 +2031,7 @@ def add_user(
 
             # Handle output mutations
             for buf in node.get_outputs():
-                # a node will mutate either 0 or 1 buffers
```

Comment:
This definitely isn't right. We mark mutations of multiple nodes through MutationOutput I believe. I think the issue is that you shouldn't be setting `get_mutation_names` on the `ir.Scan` node.

Reply:
Lol, this is so confusing. Yeah, i probably shouldn't set get_mutation_names and follow more strictly the code in UserDefinedTritonKernel, which didn't have that method. Let me try it.